WIP New landing page with a two column layout

willingc opened the pull request:
New landing page.

Fixes many warnings on doc build that was due to the recent migration of the IPython wiki to this site.

takluyver commented:
For reference, here's what the docs look like with this PR: http://test-jupyter.readthedocs.org/en/landing-twocol/

I think this is a good start, but it still feels like a lot of information that's not clearly prioritised - e.g. I don't think the docs need prominent links 'up' to the Jupyter and IPython websites, or to numfocus. Let me know if you want me to have a go at rearranging stuff and send you a PR against this.

willingc commented:
@takluyver Here's another landing page with some things moved around.
Dropped the resources to the bottom of the page; added more captions to the top section. The full table of contents is largely unchanged since I want to refactor that part as a section by section PR, i.e. the contributor guide table of contents and docs will be revised at the same time. 

<img width="1309" alt="screen shot 2016-04-06 at 11 31 06 am" src="https://cloud.githubusercontent.com/assets/2680980/14328511/dbc62dca-fbeb-11e5-9ecf-af942367edd0.png">

I still need to drop in all the links for the individual projects and their documentation, but will do that once we find a happy landing page. I'm going to lunch for a bit; feel free to send a pr to my branch or suggestions for me to integrate. Thanks!

Carreau commented on docs/source/development_guide/sphinx_directive.rst:
```diff
@@ -339,7 +335,7 @@ Functions definitions are correctly parsed
-.. ipython:: python
+.. code:: python
 
    print square(3)
```
Oh ! Some Python 2 !

Carreau commented on docs/source/index.rst:
```diff
@@ -1,72 +1,131 @@
+      <p><a href="https://test-jupyter.readthedocs.org/en/landing-twocol/development_guide/index.html">IPython Development guide</a><br/>
+         <span class="linkdescr">Legacy IPython wiki reference documentation</span></p>
+    </td></tr>
+  </table>
```
I'm not sure we want to go this route of having raw Html in our docs, as it' a rabbit hole, and it will start not working great on small screen. I'm fine if there is consensus to do it, but I'm not for it.

Carreau commented:
not favorable in having 2 columns, and especially if raw HTML. it looks great only if you have an expensive computer & and huge screen & only have the documentation on full screen. As soon as you want to tile your windows it becomes annoying .

<img width="536" alt="screen shot 2016-04-06 at 13 26 00" src="https://cloud.githubusercontent.com/assets/335567/14331708/4854963e-fbfb-11e5-855f-b1f07db80460.png">

Plus html is painful to edit. 

That's just my personal opinion.

takluyver commented:
I definitely don't think you need a huge screen or a fancy computer for the two column layout to be better. I prefer it down to about 900-1000 pixels browser width, which is absolutely reasonable on even cheap laptop screens from ten years ago.

However, it probably should be responsive so that it drops to a single column on small screens. I'm not wild on maintaining our own HTML either, but that's [what Python does too](https://hg.python.org/cpython/file/tip/Doc/tools/templates/indexcontent.html).

willingc commented:
I suppose that I can try making it a restructuredText table, but it may look weird. Alternatively, I suppose we can do one column. I suspect either way this is a short term solution until summer when hopefully we'll have some designers to work on a proper landing page.

As for Python 2 and links, I only fixed what I needed from the IPython wiki (that Jess moved) to get Sphinx to build and linkcheck without warnings. 

I'll post screenshots of a few other options.

willingc commented:
One column, no HTML screenshots

<img width="1206" alt="screen shot 2016-04-06 at 2 24 58 pm" src="https://cloud.githubusercontent.com/assets/2680980/14333712/f139be88-fc04-11e5-82f1-ddbade04682d.png">

<img width="1278" alt="screen shot 2016-04-06 at 2 25 21 pm" src="https://cloud.githubusercontent.com/assets/2680980/14333729/009d9124-fc05-11e5-9a6d-64847413d335.png">

takluyver commented:
Can we lose the sentence just beneath the title? It seems superfluous, and "official" in inverted commas looks rather strange.

willingc commented:
Using rst csv tables; no HTML

<img width="1252" alt="screen shot 2016-04-06 at 3 46 53 pm" src="https://cloud.githubusercontent.com/assets/2680980/14335293/078792c8-fc0f-11e5-8feb-89f1793ac1be.png">

<img width="1244" alt="screen shot 2016-04-06 at 3 47 02 pm" src="https://cloud.githubusercontent.com/assets/2680980/14335310/14e89ab6-fc0f-11e5-9845-507afaf4ad9f.png">

willingc commented:
Pick your favorites @takluyver @Carreau @ellisonbg for the short term landing page.

takluyver commented:
Personally, I still like the two column one. If we don't want to maintain HTML, I prefer the single column one. I don't think the tables look good.

minrk commented:
If it's as @willingc says a short-term solution, I'd go with no-html, single column. Aesthetically, though, I like the two-column HTML best.

Carreau commented:
Ok, let's debate h 2 vs one column separately. No need to bikeshed on that. This is improvment on docs. Merging.

willingc commented:
Thanks @Carreau. I'm still thinking about how to get two columns without the html.  :cake:
